tests/test_password_hasher.py

```diff
@@ -194,3 +199,35 @@ def test_params_on_wasm(self, machine):
             hash = ph.hash("hello")
 
             assert ph.verify(hash, "hello") is True
+
+
+def test_multithreaded_hashing():
+    hasher = PasswordHasher(parallelism=2)
+
+    num_passwords = 100
+
+    passwords = [secrets.token_urlsafe(15) for _ in range(num_passwords)]
+
+    def closure(b, passwords):
+        b.wait()
+        for password in passwords:
+            assert hasher.verify(hasher.hash(password), password)
+
+    max_workers = 4
+
+    chunks = [passwords[i::max_workers] for i in range(max_workers)]
+    orig_interval = sys.getswitchinterval()
+
+    with ThreadPoolExecutor(max_workers=max_workers) as tpe:
+        barrier = threading.Barrier(max_workers)
+        futures = []
+        try:
+            sys.setswitchinterval(0.00001)
+            for chunk in chunks:
+                futures.append(tpe.submit(closure, barrier, chunk))  # noqa: PERF401
```

Comment:
If I use a list comprehension like the lint wants me to, then I can't catch the case the `finally` block is trying to catch where for whatever reason spawning a thread fails. That can happen due to resource limits.

Reply:
to be clear: you mean because you wouldn't have a partial list?